Kernel/Multiboot.h

```diff
@@ -43,7 +43,12 @@ struct multiboot_mmap_entry {
     u64 addr;
     u64 len;
     u32 type;
+#if ARCH(AARCH64)
+    // __attribute__((packed)) causes alignment issues on aarch64
```

Comment:
This can probably be fixed by either trying to add attribute `__aligned__` to the temp multiboot header stack variable you created, or storing it in a function or file-static variable.

Reply:
> This can probably be fixed by either trying to add attribute `__aligned__` to the temp multiboot header stack variable you created, or storing it in a function or file-static variable.

It throws an exception when using aligned (just like it does without it).
Not sure what you mean by `storing it in a function or file-static variable`. I've tried moving it outside the init function scope which also doesn't work (if that is what you meant).

Reply:
I meant like

```
static alignas(16) multiboot_memory_map_t mmap[] = {
        { sizeof(struct multiboot_mmap_entry) - sizeof(u32),
            (u64)0x0,
            (u64)0xA2F000,
            MULTIBOOT_MEMORY_AVAILABLE }
    };
```
either at function scope, or file scope.

Reply:
> I meant like
> 
> ```
> static alignas(16) multiboot_memory_map_t mmap[] = {
>         { sizeof(struct multiboot_mmap_entry) - sizeof(u32),
>             (u64)0x0,
>             (u64)0xA2F000,
>             MULTIBOOT_MEMORY_AVAILABLE }
>     };
> ```
> 
> either at function scope, or file scope.

Still crashing due to alignment issues. Don't know enough about low-level `aarch64` to figure out exactly why packed causes this.

Reply:
:shrug:  all right, I'm sure we'll figure it out eventually